### PR TITLE
Scrypt PoW

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

**Comment by patricklodder** [comment]
need to add `test/scrypt_tests.cpp` to `src/Makefile.test.include`

File: src/test/scrypt_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include <boost/test/unit_test.hpp>
+
+#include "uint256.h"
+#include "util.h"
```
Comment (patricklodder) [inline src/test/scrypt_tests.cpp]:
```
test/scrypt_tests.cpp: In member function ‘void scrypt_tests::scrypt_hashtest::test_method()’:
test/scrypt_tests.cpp:22:42: error: ‘ParseHex’ was not declared in this scope
         inputbytes = ParseHex(inputhex[i]);
                                          ^
test/scrypt_tests.cpp:29:85: error: ‘BEGIN’ was not declared in this scope
         scrypt_1024_1_1_256_sp_generic((const char*)&inputbytes[0], BEGIN(scrypthash), scratchpad);
                                                                                     ^
Makefile:6184: recipe for target 'test/test_test_bitcoin-scrypt_tests.o' failed
```

change this into: `#include "utilstrencodings.h"` and the magic sparks shoot all over the place

**Comment by rnicoll** [comment]
Changes applied

**Comment by patricklodder** [comment]
works now. thx!

**Pull request merged**